### PR TITLE
soc: arm: nxp: add lpc55x26 support

### PR DESCRIPTION
**Pull request by maximevince**
Add basic support for the LPC55x26 SoCs

Signed-off-by: Maxime Vincent <maxime@veemax.be>

### CODE REVIEW

**Review by tmon-nordic — DISMISSED** [review]
USB part seems fine, just a single extra define for target using existing pattern.

File: soc/nxp/lpc/lpc55xxx/Kconfig.defconfig [hunk 1]
```diff
@@ -58,6 +58,14 @@ endchoice
 
 endif # SOC_LPC55S16
 
+if SOC_LPC55S26
+
+choice USB_MCUX_CONTROLLER_TYPE
+	default USB_DC_NXP_LPCIP3511
+endchoice
+
+endif # SOC_LPC55S28
```
Comment (MarkWangChinese) [inline soc/nxp/lpc/lpc55xxx/Kconfig.defconfig:67]: SOC_LPC55S26？

Comment (maximevince) [reply soc/nxp/lpc/lpc55xxx/Kconfig.defconfig]: Laser eyes!
Fixed.

File: drivers/usb/device/usb_dc_mcux.c [hunk 2]
```diff
@@ -78,7 +78,7 @@ BUILD_ASSERT(NUM_INSTS <= 1, "Only one USB device supported");
 /* Controller ID is for HAL usage */
 #if defined(CONFIG_SOC_SERIES_IMXRT5XX) || \
 	defined(CONFIG_SOC_SERIES_IMXRT6XX) || \
-	defined(CONFIG_SOC_LPC55S28) || \
+	defined(CONFIG_SOC_LPC55S26) || defined(CONFIG_SOC_LPC55S28) || \
 	defined(CONFIG_SOC_LPC55S16)
```
Comment (dleach02) [inline drivers/usb/device/usb_dc_mcux.c:82]: Not sure what clang-format is complaining about here.

**Review by dleach02 — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Pull request merged**